packages/waku/src/lib/plugins/vite-plugin-fs-router-typegen.ts

```diff
@@ -120,13 +120,14 @@ export const fsRouterTypegenPlugin = (opts: { srcDir: string }): Plugin => {
         const file = readFileSync(pagesDir + filePath).toString();
 
         return (
-          file.includes('const getConfig') ||
-          file.includes('function getConfig')
+          file.includes('const getConfig =') ||
+          file.includes('function getConfig(')
```

Comment:
wow, this looks too naive. i wonder what's the best practice with typical vite plugins.

Reply:
I completely agree. I was looking into trying to get an AST on demand of each file and I couldn't figure that out. 

It seemed like in vite, the AST is made available on the transform step, but I couldn't find a function to just call and pass text of a file. 

I think this is something we could add babel to achieve, but we also want to avoid adding dependencies, so I hesitated to do that.

But I believe an AST of each file would be the correct solution for checking for named exports.

Reply:
Vite does export a function called parseAST which comes from rollup.

I was not able to get this to work and from searching around, it seemed undocumented even in the rollup docs.

Reply:
Hm, it should be something simpler.
How about evaluating the file content with node:vm?
https://nodejs.org/api/vm.html
I'm totally unsure if it's feasible.

Reply:
otherwise, we can just use swc for now as we do in other plugins.

Reply:
vm doesn't seem to work. It seems like `SourceTextModule` is the closest to what we'd need, but as far as I can tell, it does not have support for working with typescript or jsx and we will need both.

Trying swc now...

Reply:
swc works! 🥳 Thanks for the recommendation!!

I'll update with a fix for that check shortly now :)

Reply:
fixed!